src/modules/dashboard/components/ColonyHome/ColonyMembers/ColonyMembers.tsx

```diff
@@ -22,64 +24,68 @@ const MSG = defineMessages({
     id: 'dashboard.ColonyHome.ColonyMembers.loadingData',
     defaultMessage: 'Loading members information...',
   },
+  reputationFetchFailed: {
+    id: 'dashboard.ColonyHome.ColonyMembers.reputationFetchFailed',
+    defaultMessage: "Failed to fetch the colony's members",
+  },
 });
 
 interface Props {
   colony: Colony;
+  currentDomainId?: number;
+  maxAvatars?: number;
 }
 
-const UserAvatar = HookedUserAvatar({ fetchUser: false });
-const MAX_AVATARS = 15;
+const UserAvatar = HookedUserAvatar({ fetchUser: true });
 
 const displayName = 'dashboard.ColonyHome.ColonyMembers';
 
-const ColonyMembers = ({ colony: { colonyAddress, colonyName } }: Props) => {
-  /*
-   * @TODO This will most likely be replaces with a request to the reputation
-   * oracle, in order to fetch the users sorted by reputation
-   */
+const ColonyMembers = ({
+  colony: { colonyAddress, colonyName },
+  currentDomainId = COLONY_TOTAL_BALANCE_DOMAIN_ID,
+  maxAvatars = 15,
+}: Props) => {
   const {
-    data: colonySubscribedUsers,
-    loading: loadingColonySubscribedUsers,
-  } = useColonySubscribedUsersQuery({
+    data: members,
+    loading: loadingColonyMembersWithReputation,
+  } = useColonyMembersWithReputationQuery({
     variables: {
       colonyAddress,
+      domainId: currentDomainId,
     },
   });
 
+  const membersPageRoute = useMemo(() => {
+    const baseRoute = `/colony/${colonyName}/members`;
+    if (currentDomainId === COLONY_TOTAL_BALANCE_DOMAIN_ID) {
+      return baseRoute;
+    }
+    return `${baseRoute}/${currentDomainId}`;
+  }, [currentDomainId, colonyName]);
+
```

Comment:
i feel that we don't need useMemo here, its quite inexpensive operation, we're just returning correct string

Reply:
Ok